src/client/transporter.js

```diff
@@ -53,10 +58,11 @@ class Transporter {
     this.emit('debug::response', response); // Emit after the request
     let result = {};
     if (
-      response.status !== 204 &&
-      response.headers.get('content-type')?.includes('application/json')
+
+      responseGetStatus(response) !== 204 &&
+        responseHeaderRetrieve(response, 'content-type')?.includes('application/json')
     ) {
-      result = await response.json();
+      result = await responseBodyJSON(response);
```

Comment:
I wonder if the the above check is enough to ensure there are no parsing errors in this function. ie: Do we miss a `catch`?